Replacing Companion with INSTANCE and changing the access modifier on LOGGER from private to public causes the following crash on runtime:
` org.quiltmc.loader.impl.FormattedException: org.quiltmc.loader.impl.entrypoint.QuiltEntrypointException: Exception while loading entries for entrypoint 'init' provided by 'example_mod'
	at org.quiltmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:425) ~[quilt-loader-0.17.6.jar:?]
	at org.quiltmc.loader.impl.launch.knot.Knot.launch(Knot.java:76) [quilt-loader-0.17.6.jar:?]
	at org.quiltmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:24) [quilt-loader-0.17.6.jar:?]
	at net.fabricmc.devlaunchinjector.Main.main(Main.java:86) [dev-launch-injector-0.2.1+build.8.jar:?]
Caused by: org.quiltmc.loader.impl.entrypoint.QuiltEntrypointException: Exception while loading entries for entrypoint 'init' provided by 'example_mod'
	at org.quiltmc.loader.impl.entrypoint.EntrypointStorage.lambda$getEntrypointContainers$1(EntrypointStorage.java:222) ~[quilt-loader-0.17.6.jar:?]
	at org.quiltmc.loader.impl.entrypoint.EntrypointContainerImpl.getEntrypoint(EntrypointContainerImpl.java:51) ~[quilt-loader-0.17.6.jar:?]
	at net.minecraft.Bootstrap.handler$zjp000$onInitialize(Bootstrap.java:1042) ~[minecraft-project-@-merged-named.jar:?]
	at net.minecraft.Bootstrap.initialize(Bootstrap.java:69) ~[minecraft-project-@-merged-named.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:161) ~[minecraft-project-@-merged-named.jar:?]
Caused by: org.quiltmc.loader.impl.entrypoint.QuiltEntrypointException: Exception while loading entries for entrypoint 'init' provided by 'example_mod'

	at net.minecraft.client.main.Main.main(Main.java:51) ~[minecraft-project-@-merged-named.jar:?]
	at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:577) ~[?:?]
	at org.quiltmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:423) ~[quilt-loader-0.17.6.jar:?]
	... 3 more
Caused by: org.quiltmc.loader.api.LanguageAdapterException: java.lang.IllegalAccessException: class org.quiltmc.loader.impl.util.DefaultLanguageAdapter cannot access a member of class com.example.example_mod.ExampleMod with modifiers "private"
	at org.quiltmc.loader.impl.util.DefaultLanguageAdapter.create(DefaultLanguageAdapter.java:61) ~[quilt-loader-0.17.6.jar:?]
	at org.quiltmc.loader.impl.entrypoint.EntrypointStorage$NewEntry.getOrCreate(EntrypointStorage.java:112) ~[quilt-loader-0.17.6.jar:?]
	at org.quiltmc.loader.impl.entrypoint.EntrypointStorage.lambda$getEntrypointContainers$1(EntrypointStorage.java:220) ~[quilt-loader-0.17.6.jar:?]
	at org.quiltmc.loader.impl.entrypoint.EntrypointContainerImpl.getEntrypoint(EntrypointContainerImpl.java:51) ~[quilt-loader-0.17.6.jar:?]
Caused by: org.quiltmc.loader.api.LanguageAdapterException: java.lang.IllegalAccessException: class org.quiltmc.loader.impl.util.DefaultLanguageAdapter cannot access a member of class com.example.example_mod.ExampleMod with modifiers "private"

	at net.minecraft.Bootstrap.handler$zjp000$onInitialize(Bootstrap.java:1042) ~[minecraft-project-@-merged-named.jar:?]
	at net.minecraft.Bootstrap.initialize(Bootstrap.java:69) ~[minecraft-project-@-merged-named.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:161) ~[minecraft-project-@-merged-named.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:51) ~[minecraft-project-@-merged-named.jar:?]
	at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:577) ~[?:?]
	at org.quiltmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:423) ~[quilt-loader-0.17.6.jar:?]
	... 3 more
Caused by: java.lang.IllegalAccessException: class org.quiltmc.loader.impl.util.DefaultLanguageAdapter cannot access a member of class com.example.example_mod.ExampleMod with modifiers "private"
Caused by: java.lang.IllegalAccessException: class org.quiltmc.loader.impl.util.DefaultLanguageAdapter cannot access a member of class com.example.example_mod.ExampleMod with modifiers "private"

	at jdk.internal.reflect.Reflection.newIllegalAccessException(Reflection.java:394) ~[?:?]
	at java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:674) ~[?:?]
	at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:492) ~[?:?]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:483) ~[?:?]
	at org.quiltmc.loader.impl.util.DefaultLanguageAdapter.create(DefaultLanguageAdapter.java:58) ~[quilt-loader-0.17.6.jar:?]
	at org.quiltmc.loader.impl.entrypoint.EntrypointStorage$NewEntry.getOrCreate(EntrypointStorage.java:112) ~[quilt-loader-0.17.6.jar:?]
	at org.quiltmc.loader.impl.entrypoint.EntrypointStorage.lambda$getEntrypointContainers$1(EntrypointStorage.java:220) ~[quilt-loader-0.17.6.jar:?]
	at org.quiltmc.loader.impl.entrypoint.EntrypointContainerImpl.getEntrypoint(EntrypointContainerImpl.java:51) ~[quilt-loader-0.17.6.jar:?]
	at net.minecraft.Bootstrap.handler$zjp000$onInitialize(Bootstrap.java:1042) ~[minecraft-project-@-merged-named.jar:?]
	at net.minecraft.Bootstrap.initialize(Bootstrap.java:69) ~[minecraft-project-@-merged-named.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:161) ~[minecraft-project-@-merged-named.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:51) ~[minecraft-project-@-merged-named.jar:?]
	at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:577) ~[?:?]
	at org.quiltmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:423) ~[quilt-loader-0.17.6.jar:?]
	... 3 more
`

--- a/src/main/java/com/example/example_mod/mixin/TitleScreenMixin.java
+++ b/src/main/java/com/example/example_mod/mixin/TitleScreenMixin.java
@@ -11,6 +11,6 @@ import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 public class TitleScreenMixin {
 	@Inject(method = "init", at = @At("TAIL"))
 	public void exampleMod$onInit(CallbackInfo ci) {
-		ExampleMod.Companion.getLOGGER().info("This line is printed by an example mod mixin!");
+		ExampleMod.INSTANCE.getLOGGER().info("This line is printed by an example mod mixin!");
 	}
 }

--- a/src/main/kotlin/com/example/example_mod/ExampleMod.kt
+++ b/src/main/kotlin/com/example/example_mod/ExampleMod.kt
@@ -6,7 +6,7 @@ import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 
 object ExampleMod : ModInitializer {
-    private val LOGGER: Logger = LoggerFactory.getLogger("Example Mod")
+    val LOGGER: Logger = LoggerFactory.getLogger("Example Mod")
 
     override fun onInitialize(mod: ModContainer) {
         LOGGER.info("Hello Quilt world from {}!", mod.metadata()?.name())

--- a/src/main/kotlin/com/example/example_mod/ExampleMod.kt
+++ b/src/main/kotlin/com/example/example_mod/ExampleMod.kt
@@ -5,11 +5,10 @@ import org.quiltmc.qsl.base.api.entrypoint.ModInitializer
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 
-class ExampleMod : ModInitializer {
+object ExampleMod : ModInitializer {
+    private val LOGGER: Logger = LoggerFactory.getLogger("Example Mod")
+
     override fun onInitialize(mod: ModContainer) {
         LOGGER.info("Hello Quilt world from {}!", mod.metadata()?.name())
-    }
-    companion object {
-        private val LOGGER: Logger = LoggerFactory.getLogger("Example Mod")
     }
 }

--- a/src/main/kotlin/com/example/example_mod/ExampleMod.kt
+++ b/src/main/kotlin/com/example/example_mod/ExampleMod.kt
@@ -10,6 +10,6 @@ class ExampleMod : ModInitializer {
         LOGGER.info("Hello Quilt world from {}!", mod.metadata()?.name())
     }
     companion object {
-        val LOGGER: Logger = LoggerFactory.getLogger("Example Mod")
+        private val LOGGER: Logger = LoggerFactory.getLogger("Example Mod")
     }
 }

--- a/src/main/resources/quilt.mod.json
+++ b/src/main/resources/quilt.mod.json
@@ -19,7 +19,10 @@
     },
     "intermediate_mappings": "net.fabricmc:intermediary",
     "entrypoints": {
-      "init": "com.example.example_mod.ExampleMod"
+      "init": {
+				"adapter": "kotlin",
+				"value": "com.example.example_mod.ExampleMod"
+			}
     },
     "depends": [
       {

